hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OFSPath.java

```diff
@@ -66,7 +70,13 @@
   static final String OFS_MOUNT_TMP_VOLUMENAME = "tmp";
 
   OFSPath(Path path) {
-    String pathStr = path.toUri().getPath();
+    URI uri = path.toUri();
+    // scheme is case-insensitive
+    if (!uri.getScheme().toLowerCase().equals(OZONE_OFS_URI_SCHEME)) {
+      throw new ParseException("Can't parse schemes other than ofs://.");
+    }
+    authority = uri.getAuthority();
```

Comment:
Should we set authority in the initOFSPath??
As there we have checked for auth.

Reply:
`initOFSPath` was designed to take a path without scheme or authority though. I didn't intend to change the parsing logic too much but I might do a refactoring to make it more consistent overall since you mentioned.